server/src/main/java/org/elasticsearch/tasks/TaskResult.java

```diff
@@ -174,21 +174,17 @@ public XContentBuilder innerToXContent(XContentBuilder builder, Params params) t
         return builder;
     }
 
-    public static final ConstructingObjectParser<TaskResult, Void> PARSER = new ConstructingObjectParser<>(
-            "stored_task_result", a -> {
-                int i = 0;
-                boolean completed = (boolean) a[i++];
-                TaskInfo task = (TaskInfo) a[i++];
-                BytesReference error = (BytesReference) a[i++];
-                BytesReference response = (BytesReference) a[i++];
-                return new TaskResult(completed, task, error, response);
-            });
+    public static final InstantiatingObjectParser<TaskResult, Void> PARSER;
+
     static {
-        PARSER.declareBoolean(constructorArg(), new ParseField("completed"));
-        PARSER.declareObject(constructorArg(), TaskInfo.PARSER, new ParseField("task"));
+        InstantiatingObjectParser.Builder<TaskResult, Void> builder = InstantiatingObjectParser.builder(
```

Comment:
Maybe call it `parser`? I don't feel too bad doing things like that because when it is built it become `PARSER`.